File: backend/models/Item.py
from database.base import Base
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship



class Item(Base):
 
    __tablename__ = "inventario"
 
    id           = Column(Integer, primary_key=True, index=True)
    personaje_id = Column(Integer, ForeignKey("personajes.id"), nullable=False)
 
    nombre      = Column(String(100))
    cantidad    = Column(Integer, default=1)
    peso        = Column(Integer, default=0)   # para calcular carga total — faltaba esto
    descripcion = Column(String(500))          # 100 era poco
 
    personaje = relationship("Personaje", back_populates="inventario")

File: frontend/lib/models/PersonajeComplementos/Item.dart
class Item {
  int? id;
  String nombre;
  int cantidad;
  int peso;
  String descripcion;
 
  Item({
    this.id,
    required this.nombre,
    this.cantidad = 1,
    this.peso = 0,
    this.descripcion = "",
  });
 
  factory Item.fromJson(Map<String, dynamic> json) => Item(
        id:          json['id'],
        nombre:      json['nombre'],
        cantidad:    json['cantidad']    ?? 1,
        peso:        json['peso']        ?? 0,
        descripcion: json['descripcion'] ?? "",
      );
 
  Map<String, dynamic> toJson() => {

        'nombre':      nombre,
        'cantidad':    cantidad,
        'peso':        peso,
        'descripcion': descripcion,
        
      };
}